### ISSUE TITLE
Add rule "max-props-per-line" (or something like that)

### ISSUE THREAD

**Issue by SamuelBolduc** [eslint/eslint#4369]
Based on this : https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-max-props-per-line.md

Could we have this rule directly into eslint, for Javascript in general (and not only for JSX)? It think it would be a great addition to enforce consistent code-style.

What do you think? I don't think I need to provide examples since the link I provided details the rule pretty well.

Thanks for the awesome work by the way!

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
What would this rule apply to? Object's properties only? Function arguments? Number of items in the array?

**Comment by SamuelBolduc** [comment]
I was more thinking about objects, but it could apply too to arrays and function arguments... Would it make sense to have distinct rules or all should be in the same one?

**Comment by nzakas** [comment]
@SamuelBolduc please provide some examples.

**Comment by SamuelBolduc** [comment]
Incorrect :

```javascript
const myObj = {prop1: "value1", prop2: "value2", prop3: "value3", prop4: "value4"};
```

Valid:

```javascript
const myObj = {
  prop1: "value1", 
  prop2: "value2", 
  prop3: "value3", 
  prop4: "value4"
};
```

There could be a parameter to the rule to set the max number of items on a single line maybe too!

**Comment by nzakas** [comment]
@eslint/eslint-team anyone want to champion this rule?

**Comment by gyandeeps** [comment]
@SamuelBolduc Do you want to send a PR for this rule?

**Comment by nzakas** [comment]
We still need a champion from the team for this rule before moving forward.

**Comment by SamuelBolduc** [comment]
I have tried my hands at it but I'm completely new to eslint rule development... I don't feel comfortable enough yet to champion it myself

**Comment by nzakas** [comment]
@SamuelBolduc sorry, what I meant is we need someone from the ESLint team to champion this rule for inclusion. I'm not sure it makes sense as a core rule, so I'm looking for someone to step forward and say it does.

**Comment by nzakas** [comment]
We couldn't find a champion for this rule on the ESLint team, so closing.
